data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/beacon/GetStateRootIntegrationTest.java

```diff
@@ -39,6 +45,47 @@ public void shouldReturnNoContentIfPreForkChoice_queryBySlot() throws Exception
     assertNoContent(response);
   }
 
+  @Test
+  public void shouldReturnStateRootIfBlockPresent() throws Exception {
+    startRestAPIAtGenesis();
+    createBlocksAtSlotsAndMapToApiResult(SIX);
+    final Response response = getBySlot(6);
+    assertThat(response.code()).isEqualTo(SC_OK);
+    final Bytes32 expectedRoot = getStateRootAtSlot(SIX);
+    assertThat(getBytes32FromResponseBody(response)).isEqualTo(expectedRoot);
+  }
+
+  @Test
+  public void shouldNotReturnStateRootForFutureSlot() throws Exception {
+    startRestAPIAtGenesis();
+    createBlocksAtSlotsAndMapToApiResult(SIX);
+    final Response response = getBySlot(7);
+    assertThat(response.code()).isEqualTo(SC_NOT_FOUND);
+  }
+
+  @Test
+  public void shouldReturnStateRootIfBlockMissed() throws Exception {
+    startRestAPIAtGenesis();
+    createBlocksAtSlotsAndMapToApiResult(SIX, EIGHT);
+    final Response response = getBySlot(7);
+    assertThat(response.code()).isEqualTo(SC_OK);
+    final Bytes32 expectedRoot = getStateRootAtSlot(SEVEN);
+    assertThat(getBytes32FromResponseBody(response)).isEqualTo(expectedRoot);
+  }
+
+  private Bytes32 getBytes32FromResponseBody(final Response response) throws IOException {
+    final String bytes32String = jsonProvider.jsonToObject(response.body().string(), String.class);
+    return Bytes32.fromHexString(bytes32String);
+  }
+
+  private Bytes32 getStateRootAtSlot(final UnsignedLong slot) {
+    try {
+      return combinedChainDataClient.getStateAtSlotExact(slot).get().get().hash_tree_root();
```

Comment:
nit: Probably just 
```suggestion
      return combinedChainDataClient.getStateAtSlotExact(slot).join().orElseThrow().hash_tree_root();
```

and probably then lose the try/catch.